src/clj/rems/db/test_data.clj

```diff
@@ -457,23 +457,43 @@
 
 (defn- create-draft! [user-id catids wfid field-value & [now]]
   (let [app-id (applications/create-new-draft-at-time user-id wfid (or now (time/now)))
-        _ (if (vector? catids)
-            (doseq [catid catids]
-              (db/add-application-item! {:application app-id :item catid}))
-            (db/add-application-item! {:application app-id :item catids}))
+        catids (if (vector? catids) catids [catids])
+        _ (doseq [catid catids]
+            (db/add-application-item! {:application app-id :item catid}))
         form (binding [context/*lang* :en]
-               (applications/get-form-for user-id app-id))]
+               (applications/get-form-for user-id app-id))
+        dynamic-workflow? (= :workflow/dynamic (get-in form [:application :workflow :type]))
+        save-draft-command (atom {:type :rems.workflow.dynamic/save-draft
+                                  :actor user-id
+                                  :application-id app-id
+                                  :time (get-in form [:application :start])
+                                  :items {}
+                                  :licenses {}})]
+    (when dynamic-workflow?
+      (applications/add-application-created-event! {:application-id app-id
```

Comment:
`create-new-draft-at-time` is only called from `create-draft!` and it doesn't get enough parameters for the created event. `create-draft!` is the lowest level operation that has enough context.

`create-draft!` and the rest of test data creation is also a big mess. After #852 it will be possible to delete much of it and simplify the code dramatically.

Reply:
What I object to is that this is logic in the test code. Shouldn't test code be able to call something to create a draft that adds the created event too?

Reply:
Absolutely. All the following functions should be changed to use the high-level APIs instead of using the low-level APIs from `rems.db.core`.

rems.db.test-data/create-expired-form!
rems.db.test-data/create-expired-license!
rems.db.test-data/create-workflows!
rems.db.test-data/create-resource-license!
rems.db.test-data/create-catalogue-item!
rems.db.test-data/trim-value-if-longer-than-fields-maxlength
rems.db.test-data/create-draft!
rems.db.test-data/create-application-with-expired-resource-license!
rems.db.test-data/create-application-before-new-resource-license!
rems.db.test-data/create-test-data!
rems.db.test-data/create-demo-data!

But that's outside this PR. I added it as a bullet point to #849.